src/main/java/org/elasticsearch/action/search/TransportSearchScrollAction.java

```diff
@@ -78,6 +84,34 @@ protected void doExecute(SearchScrollRequest request, ActionListener<SearchRespo
         }
     }
 
+    private void parseSource(SearchScrollRequest request) {
+        if (request.source() != null) {
```

Comment:
I think we should have a helper method that does the scroll id parsing. Now clear scroll and scroll apis have dedicated parsing logic. Even though the parsing logic is slightly differently (array vs. single value), it would be nice to reuse the parsing logic. The parsing helper method should support parsing a single scroll id and multiple scroll ids.